multiraft/multiraft.go

```diff
@@ -649,6 +649,11 @@ func (s *state) createGroup(groupID uint64) error {
 		}
 	}
 
+	// Automatically campaign and elect a leader for this group if there's
+	// exactly one known node for this group.
+	if len(cs.Nodes) == 1 && s.MultiRaft.nodeID == NodeID(cs.Nodes[0]) {
```

Comment:
This may be incorrect if we restarted while adding a second replica: we could have a config change in our log which is committed but unapplied, and we'd need to catch up and apply the log before calling an election. I'm not sure whether this is a real problem though since the second node wouldn't be able to become a leader in this state (membership changes happen one at a time so if the first node crashed while adding the second, there must not be a third). I suppose there is a problem it exists without this change too, since the ticker could fire before the rest of the log has been processed.

Another more serious problem is that Range.InitialState returns a ConfState in which this node is the only member whether we know that to be true or not. We need to populate cs.Nodes in that method only if the range is initialized.